jacoco-maven-plugin/src/org/jacoco/maven/ReportAggregateMojo.java

```diff
@@ -178,10 +181,19 @@ public String getName(final Locale locale) {
 	}
 
 	private MavenProject findProjectFromReactor(final Dependency d) {
+		VersionRange depVersionAsRange;
+		try {
+			depVersionAsRange = VersionRange.createFromVersionSpec(d.getVersion());
+		} catch (InvalidVersionSpecificationException e) {
+			throw new IllegalStateException("Could not parse the version of the dependency even though Maven" +
+					" could: " + d);
```

Comment:
Can this ever happen?

Reply:
Should not, really, but `InvalidVersionSpecificationException` is a checked exception, so there has to be something done about it...

Reply:
If I'm not mistaken, then we use `throw new AssetionError(e)` for such things - e.g. https://github.com/jacoco/jacoco/blob/ce7de98c76b8bd107c05a061193de76d21c4c205/org.jacoco.core/src/org/jacoco/core/data/ExecutionDataWriter.java#L123